File: src/main/java/com/epicsquid/ranchersdelight/machines/base/TickableEntity.java
package com.epicsquid.ranchersdelight.machines.base;

import net.minecraft.core.BlockPos;
import net.minecraft.world.level.Level;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.state.BlockState;

public interface TickableEntity {

	void tick(Level level, BlockPos pos, BlockState state);

	static <T extends BlockEntity> void tick(Level pLevel, BlockPos pPos, BlockState pState, T pBlockEntity) {
		if (pBlockEntity instanceof TickableEntity be) {
			be.tick(pLevel, pPos, pState);
		}
	}
}

File: src/main/kotlin/dev/epicsquid/ranchersdelight/machines/base/TickableEntity.kt
package dev.epicsquid.ranchersdelight.machines.base

import net.minecraft.core.BlockPos
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.state.BlockState

interface TickableEntity {
	fun tick(level: Level, pos: BlockPos, state: BlockState)

	companion object {
		fun <T : BlockEntity?> tick(level: Level, pos: BlockPos, state: BlockState, blockEntity: T) {
			if (blockEntity is TickableEntity) {
				blockEntity.tick(level, pos, state)
			}
		}
	}
}

